rackhd_BIST/test_suites.py

```diff
@@ -0,0 +1,608 @@
+import re
+import json
+import sys
+import httplib
+import pika
+import test_utils as utils
+
+CONFIGURATION = utils.CONFIGURATION
+Logger = utils.Logger()
+
+
+class RackhdServices(object):
+    """
+    RackHD services test suite
+    """
+    def __init__(self, path=None):
+        self.source_code_path = path or CONFIGURATION.get("sourceCodeRepo")
+        self.services = CONFIGURATION.get("rackhdServices")
+        self.heartbeat_unavailable_flags = self.services[:]
+        self.amqp_address = {"host": "localhost", "port": 5672}
+        self.amqp_connect_timeout = 20
+        self.amqp_connection = {}
+
+    def check_service_version(self):
+        """
+        Check RackHD version for each service
+        """
+        for service in self.services:
+            description = "Check service {} version".format(service)
+            commitStringPath = os.path.join(self.source_code_path, service, "commitstring.txt")
+            result = utils.robust_open_file(commitStringPath)
+            Logger.record_command_result(description, result, "warning")
+
+    def __operate_regular_rackhd(self, operator):
+        """
+        Start or stop RackHD services from regular RackHD code repo /var/renasar/
+        :param operator: operator for RackHD service, should be "start" or "stop"
+        """
+        for service in self.services:
+            description = "{} RackHD service {}".format(operator.capitalize(), service)
+            cmd = ["service", service, operator]
+            result = utils.robust_check_output(cmd)
+            Logger.record_command_result(description, result, 'error')
+
+    def __operate_user_rackhd(self, operator):
+        """
+        Start or stop RackHD services from user provided RackHD code repo
+        :param operator: operator for RackHD service, should be "start" or "stop"
+        """
+        if operator == "start":
+            for service in self.services:
+                description = "Start RackHD service {}".format(service)
+                os.chdir(os.path.join(self.source_code_path, service))
+                cmd = ["node index.js > /dev/null 2>&1 &"]  # RackHD services need run in background
+                result = utils.robust_check_output(cmd=cmd, shell=True)
+                Logger.record_command_result(description, result, 'error')
+        else:
+            get_process_cmd = ['ps aux | grep node| grep index.js | sed "/grep/d"| ' \
+                               'sed "/sudo/d" | awk \'{print $2}\' | sort -r -n']
+            output = utils.robust_check_output(cmd=get_process_cmd, shell=True)
+            process_list = output["message"].strip("\n").split("\n")
+            for kid in process_list:
+                description = "Stop RackHD service {}".format(kid)
+                cmd = ["kill", "-9", kid]
+                result = utils.robust_check_output(cmd)
+                Logger.record_command_result(description, result, 'error')
+
+    def __operate_rackhd_services(self, operator):
```

Comment:
The word "operate" in the function name is a little ambiguous for the reader. What about adding a word like "check" in the name?